### ISSUE TITLE
[PATCH v5.5.650] document C<use Module VERSION>

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1203]
Migrated from [rt.perl.org#2176](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=2176) (status was 'resolved')

Searchable as RT2176$

**Comment by p5pRT** [comment]
### From rmb1@cise.npl.co.uk

#### Created by rmb1@cise.npl.co.uk



The perlfunc documentation on C\<use> is not crystal clear on
&nbsp;    use Module VERSION
v\.
&nbsp;    use Module VERSION LIST
v\.
&nbsp;    use Module VERSION \(\)

So I submit the following patch&#8203;:

\*\*\* pod/perlfunc\.pod    2000/02/21 12&#8203;:06&#8203;:12     1\.1
\-\-\- pod/perlfunc\.pod    2000/02/21 18&#8203;:06&#8203;:13     1\.3
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 5140\,5145 \*\*\*\*
\-\-\- 5140\,5147 \-\-\-\-
&nbsp;  
&nbsp;  =item use Module VERSION LIST
&nbsp;  
\+ =item use Module VERSION
\+ 
&nbsp;  =item use VERSION
&nbsp;  
&nbsp;  Imports some semantics into the current package from the named module\,
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 5184\,5191 \*\*\*\*
&nbsp;  C\<use> will call the VERSION method in class Module with the given
&nbsp;  version as an argument\.  The default VERSION method\, inherited from
&nbsp;  the Universal class\, croaks if the given version is larger than the
\! value of the variable C\<$Module&#8203;::VERSION>\.  \(Note that there is not a
\! comma after VERSION\!\)
&nbsp;  
&nbsp;  Because this is a wide\-open interface\, pragmas \(compiler directives\)
&nbsp;  are also implemented this way\.  Currently implemented pragmas are&#8203;:
\-\-\- 5186\,5195 \-\-\-\-
&nbsp;  C\<use> will call the VERSION method in class Module with the given
&nbsp;  version as an argument\.  The default VERSION method\, inherited from
&nbsp;  the Universal class\, croaks if the given version is larger than the
\! value of the variable C\<$Module&#8203;::VERSION>\. 
\! Again\, there is a distinction between omitting LIST \(C\<import> called
\! with no arguments\) and an explicit empty LIST C\<\(\)> \(do not call
\! C\<import>\)\.  \(Note that there is not a comma after VERSION\!\) 
&nbsp;  
&nbsp;  Because this is a wide\-open interface\, pragmas \(compiler directives\)
&nbsp;  are also implemented this way\.  Currently implemented pragmas are&#8203;:

Robin

<details><summary>Perl Info</summary>

```


Site configuration information for perl v5.5.650:

Configured by rmb1 at Thu Feb 10 16:44:46 GMT 2000.

Summary of my perl5 (revision 5.0 version 5 subversion 650) configuration:
  Platform:
    osname=solaris, osvers=2.7, archname=sun4-solaris
    uname='sunos tempest 5.7 generic_106541-04 sun4u sparc sunw,ultra-5_10 '
    config_args='-Dcc=gcc'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
    usesocks=undef useperlio=define d_sfio=undef
    use64bits=define uselargefiles=define usemultiplicity=undef
  Compiler:
    cc='gcc', optimize='-O -g', gccversion=2.95.1 19990816 (release)
    cppflags='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include'
    ccflags ='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include -DUSE_LONG_LONG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib '
    libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
    libs=-lsocket -lnsl -ldl -lm -lc -lcrypt -lsec
    libc=/lib/libc.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-fPIC', lddlflags='-G -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.5.650:
    lib
    /home/rmb1/appl/lib/perl5/site_perl/5.005/sun4-solaris
    /home/rmb1/appl/lib/perl5/site_perl/5.005
    /home/rmb1/appl/lib/perl5/5.5.650/sun4-solaris
    /home/rmb1/appl/lib/perl5/5.5.650
    /home/rmb1/appl/lib/perl5/site_perl/5.5.650/sun4-solaris
    /home/rmb1/appl/lib/perl5/site_perl/5.5.650
    /home/rmb1/appl/lib/perl5/site_perl/5.5.640/sun4-solaris
    /home/rmb1/appl/lib/perl5/site_perl/5.5.640
    /home/rmb1/appl/lib/perl5/site_perl/5.005/sun4-solaris
    /home/rmb1/appl/lib/perl5/site_perl/5.005
    /home/rmb1/appl/lib/perl5/site_perl
    .


Environment for perl v5.5.650:
    HOME=/home/rmb1
    LANG=C
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/usr/lib
    LOGDIR (unset)
    PATH=/home/rmb1/appl/script:/usr/tempest/bin:/usr/local/bin:/usr/local/Admigration/exec:/usr/local/hotjava/bin:/usr/openwin/bin:/usr/dt/bin:/usr/ccs/bin:/usr/bin
    PERL5LIB=/home/rmb1/appl/lib/perl5/site_perl/5.005
    PERL_BADLANG (unset)
    SHELL=/bin/csh


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

>  The default VERSION method\, inherited from
>  the Universal class\, croaks if the 

s/Universal/UNIVERSAL/
